classes/Database/Install.php

```diff
@@ -104,9 +108,30 @@ public function registerHooks()
             $this->module->registerHook('displayFooterProduct') &&
             $this->module->registerHook('displayOrderConfirmation') &&
             $this->module->registerHook('actionProductCancel') &&
+            $this->module->registerHook('actionValidateOrder') &&
             $this->module->registerHook('actionOrderStatusPostUpdate') &&
             $this->module->registerHook('actionCartSave') &&
             $this->module->registerHook('displayBackOfficeHeader') &&
             $this->module->registerHook('actionCarrierProcess');
     }
+
+    /**
+     * Installs hidden tab for our ajax controller
+     *
+     * @return bool
+     */
+    public function installTab()
+    {
+        $tab = new Tab();
+        $tab->class_name = 'AdminGanalyticsAjax';
+        $tab->module = $this->module->name;
+        $tab->active = true;
+        $tab->id_parent = -1;
```

Comment:
Whoa 🤯 you're doing some nasty things here.

I understand you use this workaround to deal with a limitation of the software. Can you explain what is it? I'm curious if we can find a better cleaner solution for it

Reply:
@matks common pattern, I'm surprised it's the first time you see it

you want to have a tab to be available to perform ajax requests, but you hide it as it shouldn't be accessible normally

Reply:
> Almost every month I find something in prestashop that I never knew before XD it's so big

I didn't know also until last month Mathieu don't worry. 🤣 It seems like a hackjob but this is how it's done. You need to have every backoffice controller registered as a tab, or it won't find it and it throws error that controller not found.